Introduce FindPodDir

xpivarc opened the pull request:
### What this PR does
Current FindPodDirOnHost is actually ignoring a host all together and would try to find any Pod. FindPodDirOnHost now takes host as argument "host" and filters base on it. A FindPodDir now automatically provides the host based on "NODE_NAME" env variable.

### Special notes for your reviewer

<!-- optional -->


### Release note
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
NONE
```

xpivarc committed:
Introduce FindPodDir

Current FindPodDirOnHost is actually ignoring a host
all together and would try to find any Pod. FindPodDirOnHost
now takes host as argument "host" and filters base on it.
A FindPodDir now automatically provides the host based
on "NODE_NAME" env variable. The variable is set for
virt-handler.

Signed-off-by: Luboslav Pivarc <lpivarc@redhat.com>

dasionov reviewed (approved):
/lgtm

dankenigsberg commented on pkg/virt-handler/cmd-client/client.go:
```diff
@@ -195,15 +195,24 @@ func SocketFilePathOnHost(podUID string) string {
+}
+
+// Finds exactly one Pod dir on a host based on hostname. Returns error otherwise.
+func FindPodDirOnHost(vmi *v1.VirtualMachineInstance, host string) (string, error) {
```
Not bad! It is better to maintain a single term and not to mix host/node. (but not in this PR)

xpivarc commented on pkg/virt-handler/cmd-client/client.go:
```diff
@@ -195,15 +195,24 @@ func SocketFilePathOnHost(podUID string) string {
+func FindPodDir(vmi *v1.VirtualMachineInstance) (string, error) {
+	host, _ := os.LookupEnv("NODE_NAME")
+	return FindPodDirOnHost(vmi, host)
+}
```
I want to preserve the previous behavior, which means to pass a empty string. It is more safe and correct then returning error. For example I would need to heavily rework unit test....

xpivarc commented on pkg/virt-handler/cmd-client/client.go:
```diff
@@ -196,16 +196,25 @@ func SocketFilePathOnHost(podUID string) string {
+// Finds exactly one Pod dir on a host based on the NODE_NAME env. Returns error otherwise.
+func FindPodDir(vmi *v1.VirtualMachineInstance) (string, error) {
+	host, _ := os.LookupEnv("NODE_NAME")
+	return FindPodDirOnHost(vmi, host, SocketDirectoryOnHost)
```
`kkubevirt/pkg/virt-handler/cmd-client/client_test.go`

lyarwood replied:
`FindPodDirOnHost` has limited tests but I couldn't find a path that's testing `FindPodDir`, given both are public I'd expect some more unit tests.

xpivarc committed:
Client unit: fix FindPodDirOnHost test

The FindPodDirOnHost should not be used with File function
but rather Dir function.

Signed-off-by: Luboslav Pivarc <lpivarc@redhat.com>
